luigi/contrib/hive.py

```diff
@@ -180,8 +180,12 @@ class MetastoreClient(HiveClient):
     def table_location(self, table, database='default', partition=None):
         with HiveThriftContext() as client:
             if partition is not None:
-                partition_str = self.partition_spec(partition)
-                thrift_table = client.get_partition_by_name(database, table, partition_str)
+                try:
+                    import hive_metastore.ttypes
+                    partition_str = self.partition_spec(partition)
+                    thrift_table = client.get_partition_by_name(database, table, partition_str)
+                except hive_metastore.ttypes.NoSuchObjectException as e:
```

Comment:
My mistake. You're right. Previously there was an error, but my last commit had fixed it. I thought it still had something to do with the libraries, but if I had checked travis sooner, and seen it was this simple, I would have fixed it earlier.